src/optlang/tests/test_container.py

```diff
@@ -47,12 +47,10 @@ def test_dir(self):
         var = Variable('blub')
         self.container.append(var)
         print(dir(self.container))
-        self.assertEqual(dir(self.container),
-                         ['__contains__', '__delitem__', '__dict__', '__dir__', '__doc__', '__getattr__', '__getitem__',
-                          '__getstate__', '__init__', '__iter__', '__len__', '__module__', '__setitem__',
-                          '__setstate__', '__weakref__', '_check_for_name_attribute', '_reindex', 'append', 'blub',
-                          'clear', 'extend', 'fromkeys', 'get', 'has_key', 'items', 'iteritems', 'iterkeys',
-                          'itervalues', 'keys', 'update_key', 'values'])
+        self.assertTrue(
+            all(attribute in dir(self.container)
+                for attribute in ['__contains__', 'keys', 'values'])
+        )
```

Comment:
Could also check in the following way

```suggestion
        self.assertTrue(
            all(hasattr(self.container, attribute)
                for attribute in ['__contains__', 'keys', 'values'])
        )
```